### PR TITLE
Feature/df 1493 from release df 1505 replace x28

### PR DESCRIPTION
**Pull request by hamidsoleymani**
PLEASE REMIND that this branch was taken from ch-isco branch. That means this branch contains ch-isco changes.
This changes are due to replace x28-label to external-label. not every x28-label should be replaced.

### CODE REVIEW

File: app-external-job-ad-export-task/src/main/resources/config/application.properties [hunk 1]
```diff
@@ -33,12 +33,12 @@ spring.datasource.jobadservice.password=replace-with-password
...
-jobroom.x28.sftp.localDirectory: /tmp/batch/job-ad-x28-export
-jobroom.x28.sftp.remoteDirectory: jobroom
-jobroom.x28.sftp.xmlFileName: avamJobs_test_current.xml
-jobroom.x28.sftp.host: admin.x28.ch
-jobroom.x28.sftp.username: replace-with-username
-jobroom.x28.sftp.password: replace-with-password
+jobroom.external.sftp.localDirectory: /tmp/batch/job-ad-external-export
+jobroom.external.sftp.remoteDirectory: jobroom
+jobroom.external.sftp.xmlFileName: avamJobs_test_current.xml
+jobroom.external.sftp.host: admin.x28.ch
```
Comment (cjpsycos) [inline app-external-job-ad-export-task/src/main/resources/config/application.properties]: the host is certainly wrong

File: app-external-job-ad-import-task/src/main/resources/config/application.properties [hunk 2]
```diff
@@ -13,18 +13,19 @@ info.project.version=#project.version#
...
-jobroom.x28.sftp.remoteDirectory=jobroom
-jobroom.x28.sftp.fileNamePattern=x28Jobs_seco_current.zip
-jobroom.x28.sftp.host=admin.x28.ch
-jobroom.x28.sftp.username=replace-with-username
-jobroom.x28.sftp.password=replace-with-password
+#TODO please verify if need
+jobroom.external.sftp.localDirectory=/tmp/batch/job-ad-x28-import
+jobroom.external.sftp.remoteDirectory=jobroom
+jobroom.external.sftp.fileNamePattern=x28Jobs_seco_current.zip
+jobroom.external.sftp.host=admin.x28.ch
```
Comment (cjpsycos) [inline app-external-job-ad-import-task/src/main/resources/config/application.properties]: Host name is obsolete. change it to the new one

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/profession/ProfessionCodeType.java [hunk 3]
```diff
@@ -5,5 +5,7 @@
     SBN5,
     AVAM,
     BFS,
-    X28;
+    X28,
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/profession/ProfessionCodeType.java]: Is EXTERNAL not needed here?
Why is X28 still needed?

File: infrastructure-messagebroker-shared/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/messages/JobAdvertisementAction.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 package ch.admin.seco.jobs.services.jobadservice.infrastructure.messagebroker.messages;
 
 public enum JobAdvertisementAction {
-    APPROVE, REJECT, CANCEL, CREATE_FROM_AVAM, CREATE_FROM_X28, UPDATE_FROM_X28
+    APPROVE, REJECT, CANCEL, CREATE_FROM_AVAM, CREATE_FROM_X28, UPDATE_FROM_EXTERNAL
```
Comment (cjpsycos) [inline infrastructure-messagebroker-shared/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/messages/JobAdvertisementAction.java]: CREATE_FROM_X28 -> CREATE_FROM_EXTERNAL

File: web/src/main/resources/liquibase/master.xml [hunk 5]
```diff
@@ -43,4 +43,6 @@
     <include file="classpath:/liquibase/changelog/20190417090000_delete_task_tables.xml" relativeToChangelogFile="false"/>
     <include file="classpath:/liquibase/changelog/20190503080000_added_job_search_profile_tables.xml" relativeToChangelogFile="false"/>
     <include file="classpath:/liquibase/changelog/20190517080000_remove_geopoints_from_job_search_profile_table.xml" relativeToChangelogFile="false"/>
+    <include file="classpath:/liquibase/changelog/20190911080000_add_ch_isco_fields.xml" relativeToChangelogFile="false"/>
+    <include file="classpath:/liquibase/changelog/20191106180000_rename_to_external_job_advertisement_message_log_table.xml" relativeToChangelogFile="false"/>
```
Comment (cjpsycos) [inline web/src/main/resources/liquibase/master.xml]: Is somewhere in the database the value "X28" used and needs to be changed to "EXTERNAL"?

File: app-external-job-ad-export-task/src/main/resources/config/application.properties [hunk 6]
```diff
@@ -33,12 +33,12 @@ spring.datasource.jobadservice.password=replace-with-password
```
Comment (cjpsycos) [inline app-external-job-ad-export-task/src/main/resources/config/application.properties]: Please do not forget settings in the central config...

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/ProfessionCodeType.java [hunk 7]
```diff
@@ -4,10 +4,13 @@
 
 public enum ProfessionCodeType {
     AVAM,
+    EXTERNAL,
     X28,
```
Comment (cjpsycos) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/ProfessionCodeType.java]: Why is X28 still needed?

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/jobadvertisement/read/ElasticJobAdvertisementSearchService.java [hunk 8]
```diff
@@ -91,6 +91,8 @@
 	private static final String PATH_OCCUPATIONS_BFS_CODE = PATH_OCCUPATIONS + ".bfsCode";
 	private static final String PATH_OCCUPATIONS_SBN3_CODE = PATH_OCCUPATIONS + ".sbn3Code";
 	private static final String PATH_OCCUPATIONS_SBN5_CODE = PATH_OCCUPATIONS + ".sbn5Code";
+	private static final String PATH_OCCUPATIONS_CHISCO3_CODE = PATH_OCCUPATIONS + ".chIsco3Code";
+	private static final String PATH_OCCUPATIONS_CHISCO5_CODE = PATH_OCCUPATIONS + ".chIsco5Code";
 	private static final String PATH_X28_CODE = PATH_CTX + "jobContent.x28OccupationCodes";
```
Comment (cjpsycos) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/jobadvertisement/read/ElasticJobAdvertisementSearchService.java]: does the path change?
should also JobContent change the field to externalOccupationCodes?

File: app-external-job-ad-export-task/src/main/resources/config/application.properties (hunk 1 above)
Comment (birom) [reply app-external-job-ad-export-task/src/main/resources/config/application.properties]: We were discussing it, if we can remove the whole config from here. I think it should be only in the central config.

File: app-external-job-ad-import-task/src/main/resources/config/application.properties (hunk 2 above)
Comment (birom) [reply app-external-job-ad-import-task/src/main/resources/config/application.properties]: It should be removed and only taken from the central config.

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/profession/ProfessionCodeType.java (hunk 3 above)
Comment (hamidsoleymani) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/profession/ProfessionCodeType.java]: As I understood, we just replace those occurence of x28 which are isolated in job-ad-service. ProfessionCodeType is also used in reference-service, so we aren't replacing it at this time

File: infrastructure-messagebroker-shared/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/messages/JobAdvertisementAction.java (hunk 4 above)
Comment (hamidsoleymani) [reply infrastructure-messagebroker-shared/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/messages/JobAdvertisementAction.java]: Danke!

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/profession/ProfessionCodeType.java (hunk 3 above)
Comment (birom) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/profession/ProfessionCodeType.java]: I think the X28 codes should be further in use. We can only get rid of them after finishing this issue: https://alv-ch.atlassian.net/browse/DF-402

File: app-external-job-ad-export-task/src/main/resources/config/application.properties (hunk 6 above)
Comment (hamidsoleymani) [reply app-external-job-ad-export-task/src/main/resources/config/application.properties]: It is in this PR:
https://github.com/alv-ch/central-config/pull/127

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/ProfessionCodeType.java (hunk 7 above)
Comment (hamidsoleymani) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/ProfessionCodeType.java]: ProfessionCodeType is also used in reference-service.

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/jobadvertisement/read/ElasticJobAdvertisementSearchService.java (hunk 8 above)
Comment (hamidsoleymani) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/jobadvertisement/read/ElasticJobAdvertisementSearchService.java]: As spoke to Marton, we will keep this and wont change to external

**Comment by birom** [comment]
I have opened an other PR for this: https://github.com/alv-ch/job-ad-service/pull/411

**Pull request closed without merging**